sql/plan/alter_foreign_key.go

```diff
@@ -170,6 +170,104 @@ func (p *CreateForeignKey) String() string {
 	return pr.String()
 }
 
+// PartialResolveForeignKey is just like ResolveForeignKey except it doesn't care about refTbl or parentColumns
+// TODO: there is a lot of repeated code figure out how to write helper methods to reduce this
```

Comment:
Rather than make this a `TODO`, let's factor out the repeated code. Either make separate functions, or pass in `fkChecks` (as a `bool` preferably) and check if it's enabled before doing certain operations in `ResolveForeignKey`.